psi4/src/psi4/libfock/v.cc

```diff
@@ -2988,14 +2992,12 @@ void UV::compute_Vx(std::vector<SharedMatrix> Dx, std::vector<SharedMatrix> ret)
                 std::fill(Tap[P], Tap[P] + nlocal, 0.0);
                 std::fill(Tbp[P], Tbp[P] + nlocal, 0.0);
```

Comment:
Wouldn't a single `std::fill` be better than calling the function consecutively in a loop? If you did that, you could also use a consistent style for the screening i.e. `if (rho_a[P] + rho_b[P] < v2_rho_cutoff_) continue;`

Reply:
Probably.

`Tap` and `Tbp` are pointers to rows of a `Matrix` object, so the question is "are the columns guaranteed to be of length `nlocal`"? If so, then we're zeroing contiguous memory in a loop, and this way of doing it is silly. If not, then we need to check what the contiguous memory is and benchmark adding the overhead of zeroing additional elements versus removing the overhead of a bigger loop.

While this is a good issue to add, it's beyond the PR scope.

Reply:
Added #2846.